src/Composer/Installer.php

```
@@ -166,7 +166,7 @@ class Installer
      * @param EventDispatcher      $eventDispatcher
      * @param AutoloadGenerator    $autoloadGenerator
      */
-    public function __construct(IOInterface $io, Config $config, RootPackageInterface $package, DownloadManager $downloadManager, RepositoryManager $repositoryManager, Locker $locker, InstallationManager $installationManager, EventDispatcher $eventDispatcher, AutoloadGenerator $autoloadGenerator)
+    final public function __construct(IOInterface $io, Config $config, RootPackageInterface $package, DownloadManager $downloadManager, RepositoryManager $repositoryManager, Locker $locker, InstallationManager $installationManager, EventDispatcher $eventDispatcher, AutoloadGenerator $autoloadGenerator)
```

Why the `final` here and in Factory? This seems quite unrelated to the phpstan upgrade..

---

There is new rule that if you use `new static`, the constructor must be marked as final (or class itself).

---

Rather than marking the method as `final`, what do you think about ignoring the error for now?

---

Yeah to be honest I don't feel like this rule makes sense.. I get the point that an extending class might override the constructor *and* break the signature, but it also might do it and not break things, so this feels a little overly strict to me, especially for level1 stuff..

---

tbh it is level 0 and i dont like this rule too, should be in some strict rules or higher level